storage/store.go

```diff
@@ -154,20 +154,33 @@ func (e *rangeAlreadyExists) Error() string {
 	return fmt.Sprintf("range for Raft ID %d already exists on store", e.rng.Desc().RaftID)
 }
 
-// A RangeSlice is a slice of Range pointers used for replica lookups
-// by key.
-type RangeSlice []*Range
+// rangeKeyItem is a common interface for proto.Key and Range.
+type rangeKeyItem interface {
+	getKey() []byte
+}
+
+// rangeBtreeKey is a type alias of proto.Key that implements the
+// rangeKeyItem interface and the btree.Item interface.
+type rangeBtreeKey proto.Key
 
-// Implementation of sort.Interface which sorts by StartKey from each
-// range's descriptor.
-func (rs RangeSlice) Len() int {
-	return len(rs)
+func (k rangeBtreeKey) getKey() []byte {
+	return k
 }
-func (rs RangeSlice) Swap(i, j int) {
-	rs[i], rs[j] = rs[j], rs[i]
+
+func (k rangeBtreeKey) Less(i btree.Item) bool {
+	return bytes.Compare(k.getKey(), i.(rangeKeyItem).getKey()) < 0
 }
-func (rs RangeSlice) Less(i, j int) bool {
-	return bytes.Compare(rs[i].Desc().StartKey, rs[j].Desc().StartKey) < 0
+
+// rangeBtreeItem is a type alias of Range that implements the rangeKeyItem
+// interface and the btree.Item interface.
+type rangeBtreeItem Range
+
+func (r *rangeBtreeItem) getKey() []byte {
```

Comment:
If `getKey()` returned `proto.Key`, you could just use `r.getKey().Less(i.(rangeKeyItem).getKey())` in `Less()` below.